fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/dehardcoding/item/v1/FabricBowHooks.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.dehardcoding.item.v1;
+
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.item.ItemStack;
+
+/**
+ * An interface to implement for all custom bows.
+ *
+ */
+public interface FabricBowHooks {
+	/**
+	 * Runs all of the default arrow functions first.
+	 *
+	 * @param arrowStack        The ItemStack for the arrows
+	 * @param user              The user of the bow
+	 * @param remainingUseTicks The ticks remaining on the bow usage
+	 */
+	void onBowRelease(ItemStack arrowStack, LivingEntity user, int remainingUseTicks);
```

Comment:
I don't really know what you would do in this method? "default arrow functions" is kinda vague.

Based on the testmod I assume this would be used to prepare the projectile entity? However I see no projectile in the parameters